sw/device/lib/dif/dif_pinmux.h

```diff
@@ -0,0 +1,231 @@
+  kDifPinmuxOutselOutGpio20,
+  kDifPinmuxOutselOutGpio21,
+  kDifPinmuxOutselOutGpio22,
+  kDifPinmuxOutselOutGpio23,
+  kDifPinmuxOutselOutGpio24,
+  kDifPinmuxOutselOutGpio25,
+  kDifPinmuxOutselOutGpio26,
+  kDifPinmuxOutselOutGpio27,
+  kDifPinmuxOutselOutGpio28,
+  kDifPinmuxOutselOutGpio29,
+  kDifPinmuxOutselOutGpio30,
+  kDifPinmuxOutselOutGpio31,
+  kDifPinmuxOutselOutLast = kDifPinmuxOutselOutGpio31,
+} dif_pinmux_outsel_t;
+#define DIF_PINMUX_OUTSEL_NUM (kDifPinmuxOutselOutLast + 1)
+
+/**
+ * Pinmux peripheral inputs (MIO output to pinmux input: MIO -> peripheral).
+ * TODO - should be autogenerated?
+ */
+typedef enum dif_pinmux_peripheral_in {
+  kDifPinmuxPeripheralInGpio0 = 0,
+  kDifPinmuxPeripheralInGpio1,
+  kDifPinmuxPeripheralInGpio2,
+  kDifPinmuxPeripheralInGpio3,
+  kDifPinmuxPeripheralInGpio4,
+  kDifPinmuxPeripheralInGpio5,
+  kDifPinmuxPeripheralInGpio6,
+  kDifPinmuxPeripheralInGpio7,
+  kDifPinmuxPeripheralInGpio8,
+  kDifPinmuxPeripheralInGpio9,
+  kDifPinmuxPeripheralInGpio10,
+  kDifPinmuxPeripheralInGpio11,
+  kDifPinmuxPeripheralInGpio12,
+  kDifPinmuxPeripheralInGpio13,
+  kDifPinmuxPeripheralInGpio14,
+  kDifPinmuxPeripheralInGpio15,
+  kDifPinmuxPeripheralInGpio16,
+  kDifPinmuxPeripheralInGpio17,
+  kDifPinmuxPeripheralInGpio18,
+  kDifPinmuxPeripheralInGpio19,
+  kDifPinmuxPeripheralInGpio20,
+  kDifPinmuxPeripheralInGpio21,
+  kDifPinmuxPeripheralInGpio22,
+  kDifPinmuxPeripheralInGpio23,
+  kDifPinmuxPeripheralInGpio24,
+  kDifPinmuxPeripheralInGpio25,
+  kDifPinmuxPeripheralInGpio26,
+  kDifPinmuxPeripheralInGpio27,
+  kDifPinmuxPeripheralInGpio28,
+  kDifPinmuxPeripheralInGpio29,
+  kDifPinmuxPeripheralInGpio30,
+  kDifPinmuxPeripheralInGpio31,
+  kDifPinmuxPeripheralInLast = kDifPinmuxPeripheralInGpio31,
+} dif_pinmux_peripheral_in_t;
+#define DIF_PINMUX_PERIPHERAL_IN_NUM (kDifPinmuxPeripheralInLast + 1)
+
+/**
+ * MIO inputs (pinmux output to MIO input: peripheral -> MIO).
+ */
+typedef enum dif_pinmux_mio_in {
+  kDifPinmuxMioInPadFirst = 0,
+  kDifPinmuxMioInPadLast = DIF_PINMUX_MIO_NUM - 1,
+} dif_pinmux_mio_in_t;
+#define DIF_PINMUX_MIO_IN_NUM (kDifPinmuxMioInPadLast + 1)
+
+/**
+ * Pinmux instance state.
+ *
+ * Pinmux persistent data that is required by all the Pinmux API.
+ */
+typedef struct dif_pinmux {
+  mmio_region_t base_addr; /**< Pinmux instance base address. */
+  bool write_enabled;      /**< Configuration registers locked/unlocked. */
+} dif_pinmux_t;
+
+/**
+ * Pinmux init routine error codes.
+ */
+typedef enum dif_pinmux_init_error {
+  kDifPinmuxInitErrorNone = 0,        /**< Success. */
+  kDifPinmuxInitErrorInvalidArgument, /**< Input parameter is not valid. */
+  kDifPinmuxInitErrorRegistersLocked, /**< Peripheral is locked and cannot be
+                                         reset. */
+} dif_pinmux_init_error_t;
+
+/**
+ * Initialises an instance of Pinmux.
+ *
+ * Information that must be retained, and is required to program Pinmux, shall
+ * be stored in @p pinmux.
```

Comment:
`` ` ``-style parameter names, here and below.

Reply:
Done.